pygfx/geometries/_base.py

```diff
@@ -1,9 +1,17 @@
+from __future__ import annotations
+
+import collections
+from typing import TYPE_CHECKING, Iterable
 import numpy as np
 import pylinalg as la
 
 from ..utils.trackable import Trackable
 from ..resources import Resource, Buffer, Texture
 
+if TYPE_CHECKING:
+    import collections.abc
+    from numpy.typing import NDArray, ArrayLike
```

Comment:
Just out of curiosity, why not put all typing specific imports in this if block?

```py
from __future__ import annotations

import collections
from typing import Iterable
```

Reply:
can do if you'd like.  usually it comes from IDE autoimport and once typing is imported (for TYPE_CHECKING), things like ruff have no reason to put additional attributes from that module behind the guard

Reply:
Type hints are cool!

one other consideration is "import time". For libraries like numpy, that is fine.

But when you get to "heavy libraries" like tensorflow, or pytorch, having them be imported at runtime really slows you down.

I go through this periodically in my application, where the import time swells, then eventually i audit the code, and find that something like tensorflow / pytorch just increase the import time to 3-4 seconds.

For example, if one had a pygfx application:
```
pygfx --help
```
would be expected to be VERY fast, like 30 ms fast!

Reply:
not sure I'm following the comment.  typing needn't import heavy libraries if they're only used for type hints.  was there a suggestion/request/caution here?  or just a reminder to guard heavy imports behind `TYPE_CHECKING`?  (since this project already uses ruff, we can use [ruff-TC](https://docs.astral.sh/ruff/rules/#flake8-type-checking-tc) to ensure that

Reply:
> or just a reminder to guard heavy imports behind TYPE_CHECKING?

this.

Reply:
added TC rule to ruff linter